smtp/connection.ts

```diff
@@ -386,15 +386,27 @@ export class SMTPConnection extends EventEmitter {
 			? [codes]
 			: [250];
 
+		let hasResentForGreylist = false;
 		const response = (
 			err: Error | null | undefined,
 			msg: { code: string | number; data: string; message: string }
 		) => {
 			if (err) {
 				caller(callback, err);
 			} else {
-				if (codesArray.indexOf(Number(msg.code)) !== -1) {
+				const code = Number(msg.code);
+				if (codesArray.indexOf(code) !== -1) {
 					caller(callback, err, msg.data, msg.message);
+				} else if (
+					hasResentForGreylist === false &&
+					code === 450 &&
+					(msg.message?.toLowerCase().includes('greylist') ?? false)
```

Comment:
do you always use the nullish coaelesing as opposed to only when the left side could return false-y values?
 
since `includes` will always return boolean, using `||` would work too.

Reply:
i was being careful & short-circuiting in case `msg.message` was null; maybe that's caharp's nullable reference types poisoning my brain 😄 you're correct that it won't apply when `includes` is run.

Reply:
removed the nullish coalescing.